pkg/apis/core/types.go

```diff
@@ -5351,3 +5360,23 @@ type TopologySpreadConstraint struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector
 }
+
+// PortStatus represents the condition of a service port
+type PortStatus struct {
+	// Port is the port number of the service port of which status is recorded here
+	Port int32
+	// Protocol is the protocol of the service port of which status is recorded here
+	Protocol Protocol
+	// Error is to record any problem with the service port
```

Comment:
Actually that made me think more about it. 
I wonder if a "Status" field would be better than that "Error".  A "Status" could be set to "Enabled" or "Active" explicitly when the port is OK, so it would not be the missing "Error" that would show that state.
Another value I could imagine as an LB provider agnostic value is the "Protocol not supported".
Then there could be the "Port value is outside of the allowed port range".
And then maybe the "Not applicable with the configured set of ports"

And of course there is the option to put here any provider specific status message about the given port.

I.e. we would provide some pre-defined constant values for the most straightforward and common cases. But we would not restrict the possible set of values to those. 

Also then I think we would need a new condition? I mean, for the case when mixed protocols are supported, but there is a problem with an actual port definitions in the LB.

What do you think?

Reply:
So far we have gotten away without detailed error in the API, and I think we should try to stay there, lest we end up enumerating all sorts of hypotheticals.

What about a single condition that says "one or more ports are not successfully configured", and a per-port error (as you specced).  The error can be a condition.type style name (FooBar or company.com/specific-error).  If we need to define a SMALL set of common values, we can (e.g. MixedProtocolNotSupported).

Does that suffice?

Reply:
Yeah. So I made the changes. Let's see if I interpreted right.
- the condition is about port status in general
- as I understand PortStatus would be a place for error reports, so I added Error back as mandatory field
- I have not defined any error value for the time being. Providers can use the Error field for their own values.